tests/metrics/event_manager_test.py

```diff
@@ -644,3 +653,72 @@ async def test_bad_kafka_after_initialize(
 
     assert len(working) == 1
     assert len(started_again) == max_batch_size + 5
+
+
+@pytest.mark.asyncio
+async def test_bad_schema_manager_mid_registration(
+    resiliency_metrics_stack: MetricsStack,
+    log_output: LogCapture,
+    time_machine: TimeMachineFixture,
+) -> None:
+    event_manager = resiliency_metrics_stack.event_manager
+    registry = resiliency_metrics_stack.kafka_stack.schema_registry_container
+    backoff_interval = resiliency_metrics_stack.metrics_config.backoff_interval
+
+    # This should be a legit event publisher
+    event = await event_manager.create_publisher("myevent", MyEvent)
+
+    registry.stop()
+
+    # This should be a failed event publisher
+    other_event = await event_manager.create_publisher(
+        "myotherevent", MyOtherEvent
+    )
+
+    # These shouldn't raise exceptions, but they shouldn't make it to kafka.
```

Comment:
Am I confused?  Should this be, "these should raise exceptions" instead?  Otherwise, what shouldn't make it to Kafka, if there's no exception?

Reply:
This is testing the resiliency functionality. Because the schema manager died in the middle of event registration, trying to publish should just log error messages (once per backoff-time-window) instead of raising an exception and thus killing the app.

I changed the `but` to an `and`.